cloud/amazon/ec2_vpc_dhcp_options.py

```diff
@@ -128,105 +157,204 @@
+
+def fetch_dhcp_options_for_vpc(vpc_conn, vpc_id):
     """
     Returns the DHCP options object currently associated with the requested VPC ID using the VPC
     connection variable.
     """
-    vpcs = vpc_connection.get_all_vpcs(vpc_ids=[vpc_id])
+    vpcs = vpc_conn.get_all_vpcs(vpc_ids=[vpc_id])
     if len(vpcs) != 1:
-      return None
-    dhcp_options = vpc_connection.get_all_dhcp_options(dhcp_options_ids=[vpcs[0].dhcp_options_id])
+        return None
+    dhcp_options = vpc_conn.get_all_dhcp_options(dhcp_options_ids=[vpcs[0].dhcp_options_id])
     if len(dhcp_options) != 1:
-      return None
+        return None
     return dhcp_options[0]
 
+def match_dhcp_options(vpc_conn, tags=None, options=None):
+    """
+    Finds a DHCP Options object that optionally matches the tags and options provided
+    """
+    dhcp_options = vpc_conn.get_all_dhcp_options()
+    for dopts in dhcp_options:
+        if tags and get_resource_tags(vpc_conn, dopts.id) == tags:
+            if options and dopts.options == options:
+                return(True, dopts)
+    return(False, None)
 
-def _get_vpcs_by_dhcp_options(dhcp_options_id, vpc_connection):
-    return vpc_connection.get_all_vpcs(filters={'dhcpOptionsId': dhcp_options_id})
-
-
-def _get_updated_option(requested, existing, inherit):
-    if inherit and (not requested or requested == ['']):
-        return existing
+def remove_dhcp_options_by_id(vpc_conn, dhcp_options_id):
+    associations = vpc_conn.get_all_vpcs(filters={'dhcpOptionsId': dhcp_options_id})
+    if len(associations) > 0:
+        return False
     else:
-        return requested
-
+        vpc_conn.delete_dhcp_options(dhcp_options_id)
+        return True
 
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(dict(
-        domain_name=dict(type='str', default=''),
-        dns_servers=dict(type='list', default=[]),
-        ntp_servers=dict(type='list', default=[]),
-        netbios_name_servers=dict(type='list', default=[]),
-        netbios_node_type=dict(type='int', default=2),
-        vpc_id=dict(type='str', required=True),
+        dhcp_options_id=dict(type='str', default=None),
+        domain_name=dict(type='str', default=None),
+        dns_servers=dict(type='list', default=None),
+        ntp_servers=dict(type='list', default=None),
+        netbios_name_servers=dict(type='list', default=None),
+        netbios_node_type=dict(type='int', default=None),
+        vpc_id=dict(type='str', default=None),
         delete_old=dict(type='bool', default=True),
-        inherit_existing=dict(type='bool', default=False)
+        inherit_existing=dict(type='bool', default=False),
+        tags=dict(type='dict', default=None, aliases=['resource_tags']),
+        state=dict(type='str', default='present', choices=['present', 'absent'])
         )
     )
 
     module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
     params = module.params
+    found = False
```

Comment:
Ah, OK. It looks like the only time that that distinction is relevant is after the last usage of `found` which is what I was looking at, but I certainly see the benefit of future-proofing the code.